README.md

```diff
@@ -12,194 +12,129 @@ shared. These resources have a different lifecycle than resources in workspaces.
 low probability of a naming collision.
 
-> The Azure Resource Manager APIs can be retried if a transient error occurs - the API is idempotent. However, The
-> request must be the same as the failed one to avoid duplicating resources in the deployment. The deployment could
-> create
-> duplicate resources if the resource's name is auto-generated and changes in every request.
-
-An instance of the resource name generator is included in the deployment context.
-
-```java
- var storage=azureResourceManager
-        .storageAccounts()
-        .define(definitionContext.resourceNameGenerator().nextName(20))
-        .withRegion(Region.US_EAST2)
-        .withExistingResourceGroup(definitionContext.resourceGroup());
-
-```
-
-### Handling Prerequisites
-
-The library deploys resources in a non-deterministic order. Therefore, it is not possible to assume any specific order.
-For cases when a resource must be created before other resources, you can create a prerequisite deployment inside your
-definition.
-
-```java
+### Error handling and retrying
 
-class FooLZDefinitionV1 extends LandingZoneDefinition {
-
-    protected FooLZDefinitionV1(
-            AzureResourceManager azureResourceManager, RelayManager relayManager) {
-        super(azureResourceManager, relayManager);
-    }
-
-    @Override
-    public Deployable definition(DefinitionContext definitionContext) {
-
-        var vNet =
-                azureResourceManager
-                        .networks()
-                        .define(definitionContext.resourceNameGenerator().nextName(20))
-                        .withRegion(Region.US_EAST2)
-                        .withExistingResourceGroup(definitionContext.resourceGroup())
-                        .withAddressSpace("10.0.0.0/28")
-                        .withSubnet("subnet1", "10.0.0.0/29")
-
-        var prerequisites =
-                deployment
-                        .definePrerequisites()
-                        .withVNetWithPurpose(vNet, "subnet1", SubnetResourcePurpose.WORKSPACE_STORAGE_SUBNET)
-                        .deploy();
-
-        var storage =
-                azureResourceManager
-                        .storageAccounts()
-                        .define(definitionContext.resourceNameGenerator().nextName(20))
-                        .withRegion(Region.US_EAST2)
-                        .withExistingResourceGroup(definitionContext.resourceGroup());
-
-        return definitionContext
-                .deployment()
-                .withResourceWithPurpose(storage, ResourcePurpose.SHARED_RESOURCE);
-    }
-```
+Each step is responsible for error handling. Usually in case a step is responsible for creation of just one resource (recommended) there is no need to add additional error handling.
+But this can be changed depending on complexity/requirements. Base step from which all steps are inherited contains some common error handling as well.
+For instance base class handles situation when resource already exist.
```

Comment:
Suggested wording:

Each step is responsible for error handling. In the case that a step is responsible for creation of just one resource (recommended), there is usually no need to add additional error handling, but this can be changed depending on complexity/requirements. The base step from which all steps are inherited contains some common error handling as well; for instance, the base class handles the situation where the resource already exists.